windows/check_quiesce_snapshot/check_quiescing_type.yml

```diff
@@ -0,0 +1,99 @@
+      - vm_config.layoutEx.file | length > 0
+    fail_msg: "Not get VM files info from 'vm_config.layoutEx.file' after taking snapshot."
+
+- name: "Set fact of the list of vss_manifests zip files"
+  ansible.builtin.set_fact:
+    vm_vss_manifests_list_after: "{{ vm_config.layoutEx.file | map(attribute='name') | select('search', 'vss_manifests') }}"
+
+- name: "Display the vss_manifests zip files after taking quiesce snapshot"
+  ansible.builtin.debug: var=vm_vss_manifests_list_after
+
+- name: "Set fact of the new vss_manifests zip file"
+  ansible.builtin.set_fact:
+    vm_vss_manifests_new: "{{ vm_vss_manifests_list_after | difference(vm_vss_manifests_list_before) }}"
+
+- name: "Check one new vss_manifests zip file generated"
+  ansible.builtin.assert:
+    that:
+      - vm_vss_manifests_new | length == 1
+    fail_msg: >
+      Before taking quiesce snapshot, the vss_manifests zip files are: {{ vm_vss_manifests_list_before }},
+      after taking quiesce snapshot, the vss_manifests zip files are: {{ vm_vss_manifests_list_after }},
+      not get the new generated zip file.
+
+- name: "Set fact of local path of the new vss_manifests zip file"
+  ansible.builtin.set_fact:
+    vm_vss_manifests_path: "{{ current_test_log_folder }}/{{ vm_vss_manifests_new[0] | basename }}"
+    vm_vss_manifests_unzip_path: "{{ current_test_log_folder }}/{{ (vm_vss_manifests_new[0] | basename).split('.')[0] }}"
+
+- name: "Create local directory for unarchived files"
+  include_tasks: ../../common/create_directory.yml
+  vars:
+    dir_path: "{{ vm_vss_manifests_unzip_path }}"
+
+- name: "Get the new vss_manifests zip file to local"
+  include_tasks: ../../common/esxi_download_datastore_file.yml
+  vars:
+    src_datastore: "{{ datastore }}"
+    src_file_path: "{{ vm_vss_manifests_new[0].split(']')[1].strip() }}"
+    dest_file_path: "{{ vm_vss_manifests_path }}"
+
+- name: "Get the list of files in the new vss_manifests zip file"
+  ansible.builtin.unarchive:
+    src: "{{ vm_vss_manifests_path }}"
+    dest: "{{ vm_vss_manifests_unzip_path }}"
+    list_files: true
+  register: vm_vss_manifests_files
+
+# Windows Server 2025 VM with NVMe boot disk controller does not
+# suppport application consistent quiesce snapshot due to a known Windows issue.
+# Please keep track of the 3rd party issue status and vSphere support status.
+- name: "Get boot disk controller type in guest OS"
+  include_tasks: ../utils/win_get_boot_disk_ctl_type.yml
+  when: not new_vm
+
+- name: "Set fact of support application consistent quiescing situation"
+  ansible.builtin.set_fact:
+    winsrv_app_quiesce_support: true
+  when:
+    - guestinfo_guest_full_name | regex_search('Windows Server ([0-9]{4})')
+    - guestinfo_guest_full_name | regex_search('Windows Server ([0-9]{4})') == guest_os_ansible_distribution | regex_search('Windows Server ([0-9]{4})')
+    - (new_vm and boot_disk_controller != 'nvme') or (not new_vm and win_boot_disk_ctl_type != 'nvme')
```

Comment:
I saw above comments only mentioned `Windows Server 2025` VM with NVMe boot disk not support. But here doesn't check the Windows Server's version is 2025